profiler/src/main/java/com/splunk/opentelemetry/profiler/snapshot/AccumulatingStagingArea.java

```diff
@@ -45,9 +46,9 @@ public void stage(String traceId, StackTrace stackTrace) {
 
   @Override
   public void empty(String traceId) {
-    List<StackTrace> stackTraces = this.stackTraces.remove(traceId);
+    Queue<StackTrace> stackTraces = this.stackTraces.remove(traceId);
     if (stackTraces != null) {
-      exporter.get().export(stackTraces);
+      exporter.get().export(new ArrayList<>(stackTraces));
```

Comment:
if you made the `StackTraceExporter` accept a `Collection` than you could skip the copy.